### ISSUE TITLE
Reset macro kills binds set in-game.

### ISSUE THREAD

**Issue by xizar** [emersonrp/bindcontrol#78]
If I set -/= for "prev"/"next tray", using the reset macro kills them.

Is it tenable to leave binds that aren't controlled by Bindcontrol alone?

Is it tenable to add default keybind targets (not the key to which their bound, only the command) in BC?

(Perhaps I don't know how to make it work, but adding simplebinds for next and prev tray don't seem to work..   nothing gets bound to either - or =)

edit: okay, maybe I'm too dumb to make simple binds work..  I set up "bind contents" as "next_tray" (with and without quotes) with a bind key of "J" and nothing happens in-game.  If I set up the same bind in-game manually via "/bind j "nexttray"" it works.

**Comment by xizar** [comment]
okay, I looked at the reset.txt file and the bind isn't put in there.  (granted, I'm only assuming it'd be in there.)
[Ninhegal.bcp.txt](https://github.com/user-attachments/files/16406137/Ninhegal.bcp.txt)

**Comment by emersonrp** [comment]
(Sorry for the delay -- been traveling this week.)

So there are two approaches to this.

The first is just what you ask, there's a preference in the preferences dialog "Set binds to default on reset" that makes the reset do /keybind_reset at the start, which sets binds to the game's default.  Unchecking that will make it so anything you add outside of BindControl doesn't get reset away.  This can lead to creeping weirdness if you're in the habit of /binding things manually, but it's in there in case you want to manage things that way, and /keybind_reset is always there if manual tinkerings get problematic.

The other is to add two Simple Binds, one for each key, with "Custom Bind" steps that contain the /prev_tray etc commands.  This is a bit more fiddly but makes it so BindControl is managing everything.  It's probably reasonable to consider it a bug that prev_tray and next_tray aren't accessible in any way except as custom commands.  This might prompt me to add them to the "Gameplay" tab in or around the "Trays" section.

Let me know your thinking on this.

**Comment by xizar** [comment]
I have thoughts on using the Preferences dialogue to affect preferences for things that aren't the software itself.  I will hasten to point out that I know fuckall about UX/UI design theory and don't code stuff of any use to anyone, so with that caveat:

I never would have looked in Preferences for settings to change what binds are written.  Because of how global (all characters, all the times) something like "binds reset key" is, I get that it isn't something someone would want different for different characters.

I've done a little more tooling around in CoX, and it seems some things like "next/prev tray" are tied to "Options>Keymapping>Keybind Profile" in-game.  Resetting all keybinds and then running BC's reset macro seems to have the expected effect for everything else (I didn't go through and check everything, just randomly checked a few pages in Keymapping.)

Regarding using a custom bind: that didn't work.  BC would not write out a bind for it.  Using "showbind" in-game showed a binding of "nothing" if I try to use prev/next tray while if I set up a simple bind in BC with nonsense, that would be reflected in showbind in-game.  Manually setting the bind for next_tray in-game would work as expected.

**Comment by emersonrp** [comment]
> I have thoughts on using the Preferences dialogue to affect preferences for things that aren't the software itself.

Yeah, I totally get that.  I'm wondering if it's at least partially a question of framing -- would it feel better if the preference were called something like "Manage all binds with BindControl" or "Honor non-BindControl binds" or something?  It's not actually deciding which binds to write, it's just changing the global behavior of what reset does.  Moving it out of the Prefs dialog would yeah make it profile-specific, which as you say seems not right also.

> Regarding using a custom bind: that didn't work. BC would not write out a bind for it. Using "showbind" in-game showed a binding of "nothing" if I try to use prev/next tray while if I set up a simple bind in BC with nonsense, that would be reflected in showbind in-game. Manually setting the bind for next_tray in-game would work as expected.

Oh interesting, I need to look into that.  Do you have a profile saved where you created the custom binds but it didn't want to write them?  I haven't actually squinted at the Custom Bind innards in a long time so there might well be some cruft in there.  And I'm still warm to the idea of surfacing next-tray / prev-tray someplace in the Gameplay tab.

**Comment by xizar** [comment]
> > Regarding using a custom bind: that didn't work. BC would not write out a bind for it. Using "showbind" in-game showed a binding of "nothing" if I try to use prev/next tray while if I set up a simple bind in BC with nonsense, that would be reflected in showbind in-game. Manually setting the bind for next_tray in-game would work as expected.
> 
> Oh interesting, I need to look into that. Do you have a profile saved where you created the custom binds but it didn't want to write them? I haven't actually squinted at the Custom Bind innards in a long time so there might well be some cruft in there. And I'm still warm to the idea of surfacing next-tray / prev-tray someplace in the Gameplay tab.

That profile is the one I uploaded above.

**Comment by emersonrp** [comment]
Hmm that profile doesn't have anything in the Custom Binds section at all.  I'm wondering if the Simple Bind you worked on was somehow malformed and just didn't take at all, including not getting saved.

Regardless, though, I'm seeing what you're seeing:  when I build simple binds that should do the right thing, and they get correctly saved into the profile and correctly written into the reset.txt file, they just don't "take" when the reset file is loaded.  This is mysterious, and might be a Homecoming bug.  I'm gonna tinker a little more, but I may end up trying to report a bug upstream.

**Comment by emersonrp** [comment]
OK, curiouser and curiouser.  I tinkered with it for a little bit and then abruptly it started working, ie, loading the reset file correctly bound -/= to prev/next per the Simple Binds.  I fiddled with the Reset Binds button in-game, as well as the load/save from file buttons at the bottom of the keybinds dialog, so one or more of those steps might have jiggled something loose.  It's working for me now.  Try out the attached profile and let me know your luck.

[Ninhegal.bcp.txt](https://github.com/user-attachments/files/16488578/Ninhegal.bcp.txt)

**Comment by emersonrp** [comment]
Just made a release that adds next / prev tray buttons, for all three trays, to the Gameplay tab.  Seems to work for me.

https://github.com/emersonrp/bindcontrol/releases/tag/v0.21.5

**Comment by emersonrp** [comment]
Though the "binds in reset.txt didn't apply" spurious behavior was never explained, it doesn't seem to be doing that now.  The Prev / Next tray buttons are working as designed, as well as the "set all binds to default on reset" checkbox.  The latter might still want some language tweaks and/or documentation, but I'm closing this issue as complete.  If the spurious behavior shows up again, let's make a new issue for it.
